server/methods/catalog.js

```diff
@@ -1259,15 +1258,46 @@ Meteor.methods({
         isVisible: !product.isVisible
       }
     }, {
-      selector: {
-        type: product.type
-      }
-    });
+        selector: {
+          type: product.type
+        }
+      });
 
     // if collection updated we return new `isVisible` state
     return res === 1 && !product.isVisible;
 
     Logger.debug("invalid product visibility ", productId);
     throw new Meteor.Error(400, "Bad Request");
+  },
+  /**
+  * products/productGrids
+  * @summary Removes blank products
+  * @param {Object} [product] - optional product object
+  * @return {*} none
+  */
+  "products/removeBlankProducts": function () {
+    if (!Reaction.hasPermission("createProduct")) {
+      throw new Meteor.Error(403, "Access Denied");
+    }
+    const blankProducts = Products.find({
+      $and: [
+        { ancestors: { $size: 0 } },
+        {
+          $or: [
+            { title: "" },
+            { description: "" },
+            { vendor: "" }
+          ]
+        }
+      ]
+    }, { fields: { _id: 1 } });
```

Comment:
I think you can do a `.distinct('_id')` instead of a separate ```const blankProductsId = blankProducts.map(products => products._id);``` Can you give that a shot ans see how it goes?
Docs here: https://docs.mongodb.com/manual/reference/method/db.collection.distinct/